### PR TITLE
Decompile unk_02088DD8.s

### PR DESCRIPTION
**Pull request by SNBeast**
I'm a noob, so lemme know if I should do something different. And where did that ncgr come from?

### CODE REVIEW

File: arm9/src/unk_02088DD8.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#include "global.h"
+#include "heap.h"
+#include "pokemon.h"
+#include "unk_02088DD8.h"
+
+THUMB_FUNC struct UnkStruct_02088DD8* FUN_02088DD8(u32 r0) {
```
Comment (tgsm) [inline arm9/src/unk_02088DD8.c:6]:
```suggestion
THUMB_FUNC struct UnkStruct_02088DD8* FUN_02088DD8(u32 heap_id) {
```

File: arm9/src/unk_02088DD8.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    struct UnkStruct_02088DD8 *returnPointer = AllocFromHeap(r0, 24);
+    __builtin__clear(returnPointer, 24);
```
Comment (tgsm) [inline arm9/src/unk_02088DD8.c:8]: `sizeof` should be used instead here. `sizeof(struct UnkStruct_2088DD8)`

File: arm9/src/scrcmd_24.c [hunk 3]
```diff
@@ -10,10 +10,6 @@ extern struct UnkStruct_02037CF0* FUN_02037CF0(u32 heap_id, struct UnkSavStruct8
 extern u8 FUN_02037D5C(struct UnkStruct_02037CF0*);
 extern void FUN_02038864(struct UnkSavStruct80*, struct UnkStruct_02088DD8*);
 extern BOOL FUN_0203BC04(struct ScriptContext* ctx);
-extern struct UnkStruct_02088DD8* FUN_02088DD8(u32 heap_id);
```
Comment (red031000) [inline arm9/src/scrcmd_24.c:13]: the header file should be included in this file

File: arm9/src/unk_02088DD8.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+#define WOTBL_END           0xFFFF
+#define WOTBL_MOVE_MASK     0x01FF
+#define WOTBL_MOVE_SHIFT         0
+#define WOTBL_LVL_MASK      0xFE00
+#define WOTBL_LVL_SHIFT          9
+#define WOTBL_MOVE(x) ((u16)(((x) & WOTBL_MOVE_MASK) >> WOTBL_MOVE_SHIFT))
+#define WOTBL_LVL(x) (/*(u8)*/(((x) & WOTBL_LVL_MASK) >> WOTBL_LVL_SHIFT))
+// i don't know why either.
```
Comment (red031000) [inline arm9/src/unk_02088DD8.c:23]: ???

File: arm9/src/unk_02088DD8.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        if (tableFromFile[i] == WOTBL_END) {
+            returnTable[k] = WOTBL_END;
+            break;
+        }
+        else {
+            if (WOTBL_LVL(tableFromFile[i]) > level) continue;
+
+            tableFromFile[i] = WOTBL_MOVE(tableFromFile[i]);
+            
+            j = 0;
```
Comment (red031000) [inline arm9/src/unk_02088DD8.c:50]: this is a really weird way of doing this

File: include/unk_02088DD8.h [hunk 6]
```diff
@@ -1,6 +1,10 @@
 #ifndef POKEDIAMOND_UNK_02088DD8_H
 #define POKEDIAMOND_UNK_02088DD8_H
 
+#include "unk_020377F0.h"
+
+extern void LoadWotbl_HandleAlternateForme(int species, int forme, u16 * wotbl);
```
Comment (red031000) [inline include/unk_02088DD8.h:6]: externs should be within the c file otherwise it can cause some major weirdness

File: arm9/src/unk_02088DD8.c [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+#define WOTBL_END           0xFFFF
+#define WOTBL_MOVE_MASK     0x01FF
+#define WOTBL_MOVE_SHIFT         0
+#define WOTBL_LVL_MASK      0xFE00
+#define WOTBL_LVL_SHIFT          9
+#define WOTBL_MOVE(x) ((u16)(((x) & WOTBL_MOVE_MASK) >> WOTBL_MOVE_SHIFT))
+#define WOTBL_LVL(x) (/*(u8)*/(((x) & WOTBL_LVL_MASK) >> WOTBL_LVL_SHIFT))
+// i don't know why either.
+
+THUMB_FUNC void* GetEligibleLevelUpMoves(struct Pokemon* pokemon, u32 heap_id) {
```
Comment (red031000) [inline arm9/src/unk_02088DD8.c:25]: return type should be u16 * not void *

**Review by red031000 — CHANGES_REQUESTED** [review]
the ncgr is there because either you didn't build on the latest repo or you didn't clean after build, cleaning and committing in that order should remove it

File: arm9/src/unk_02088DD8.c (hunk 5 above)
Comment (SNBeast) [reply arm9/src/unk_02088DD8.c:50]: That's what the assembly strongly suggests, so prefer to ask GameFreak.

File: arm9/src/unk_02088DD8.c (hunk 4 above)
Comment (SNBeast) [reply arm9/src/unk_02088DD8.c:23]: The code matches WOTBL_LVL as previously defined, except that the previously defined WOTBL_LVL casts to u8 but the code in this file does not.

File: arm9/src/unk_02088DD8.c (hunk 5 above)
Comment (red031000) [reply arm9/src/unk_02088DD8.c:50]: ok but can you not include the assignment in the for loop? does it not match if you do that?

Comment (SNBeast) [reply arm9/src/unk_02088DD8.c:50]: ohh the j = 0, thought you meant the other thing. yeah that matches, sorry for not trying that sooner

File: arm9/src/scrcmd_24.c (hunk 3 above)
Comment (SNBeast) [reply arm9/src/scrcmd_24.c:13]: What change are you suggesting? There's already a `#include "unk_02088DD8.h"`.

Comment (red031000) [reply arm9/src/scrcmd_24.c:13]: ohh sorry, ignore this then

**Review by red031000 — APPROVED** [review]

**Pull request merged**